src/lesson14/library/Library.java

```diff
@@ -0,0 +1,251 @@
+            ex.printStackTrace(System.out);
+        } catch (SAXException ex) {
+            ex.printStackTrace(System.out);
+        } catch (IOException ex) {
+            ex.printStackTrace(System.out);
+        }
+        return null;
+    }
+
+    public void createListOfCustomers() {
+        List<List<String>> objects = readFromXML("src\\lesson14\\library\\customers.xml");
+        allCustomers.addAll(createCustomer(objects));
+    }
+
+    public void createLibrary() {
+        List<List<String>> objects = readFromXML("src\\lesson14\\library\\books.xml");
+        allBooks.addAll(createBook(objects));
+        createListBooksAvailableForReading();
+        createListBooksCanBeRemovedFromTheLibrary();
+    }
+
+    public void createListBooksAvailableForReading() {
+        for (Book book : allBooks) {
+            if (book.getAvailableForReading()) {
+                for (int i = 0; i < book.getAmount(); i++) {
+                    booksAvailableForReading.add(book);
+                }
+            }
+        }
+    }
+
+    public void createListBooksCanBeRemovedFromTheLibrary() {
+        for (Book book : allBooks) {
+            if (book.getOutLibrary()) {
+                booksCanBeRemovedFromTheLibrary.add(book);
+            }
+        }
+    }
+
+    public void printCustomers(List<Customer> customers) {
+        for (int i = 0; i < customers.size(); i++) {
+            System.out.println(i + ". " + customers.get(i).getName());
+        }
+    }
+
+    public void printBooks(List<Book> books) {
+        for (int i = 0; i < books.size(); i++) {
+            System.out.println(i + ". " + books.get(i).getName() + ", " + books.get(i).getAuthor() + ", " +
+                    books.get(i).getGenre() + ", " + books.get(i).getYear() + ", " + books.get(i).getCover());
+        }
+    }
+
+    public List<Book> getAllBooks() {
+        return allBooks;
+    }
+
+    public void setAllBooks(List<Book> allBooks) {
+        this.allBooks = allBooks;
+    }
+
+    public List<Book> getBooksAvailableForReading() {
+        return booksAvailableForReading;
+    }
+
+    public void setBooksAvailableForReading(List<Book> booksAvailableForReading) {
+        this.booksAvailableForReading = booksAvailableForReading;
+    }
+
+    public List<Book> getBooksCanBeRemovedFromTheLibrary() {
+        return booksCanBeRemovedFromTheLibrary;
+    }
+
+    public void setBooksCanBeRemovedFromTheLibrary(List<Book> booksCanBeRemovedFromTheLibrary) {
+        this.booksCanBeRemovedFromTheLibrary = booksCanBeRemovedFromTheLibrary;
+    }
+
+    public List<Customer> getAllCustomers() {
+        return allCustomers;
+    }
+
+    public void setAllCustomers(List<Customer> allCustomers) {
+        this.allCustomers = allCustomers;
+    }
+
+    public void takeBookOutLibraryOrNot(Customer customer, Book book) {
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Do you want to take it out library or to the reading room?");
```

Comment:
not clear what to enter as response

Reply:
0 - out library, 1 - to the reading room. I added an explanation to the code